### ISSUE TITLE
[Bug] If you have an existing ID that you restore, you cannot create a new, free sponsored name

### ISSUE THREAD

**Issue by jehunter5811** [blockstack/blockstack-browser#1449]
Steps to reproduce: 

1) From either a local download of the most recent Blockstack Browser or from browser.blockstack.org, restore from an existing keychain. 
2) If that restore was for an ID that already had a username, create a new username. 
3) If that restore was for an ID that did not have a username yet, try to add one. 

In both cases, the user is unable to add a free, sponsored username. This functionality appears to only apply to brand new Blockstack users, not those who have existing IDs.

**Comment by yknl** [comment]
Hey @jehunter5811 this is a feature that we are looking to push out soon. In fact, there is already code in the browser that allows you register sponsored names with an existing ID. We need to do more testing before we can flip the switch.

**Comment by wbobeirne** [comment]
Just realized #1499 is a duplicate of this. However, the bulk of the discussion has moved over there, so I'm going to close this issue rather than vice versa.
